### PR TITLE
Homework 3 done

### PR DESCRIPTION
**Pull request by goharminasyan**

### CODE REVIEW

File: src/arrays/ArrayUtilTest.java [hunk 1]
```diff
@@ -4,5 +4,17 @@
...
+                   ArrayUtil.createArray(4);
+                  ArrayUtil.create2dArray(2,3);
+                 ArrayUtil.addFirst(new int [] {5,4,8,1}, 2);
+                ArrayUtil.contains(new int [] {0,4,5,6,8}, 5);
+               ArrayUtil.indexOf(new int [] {25,1,45,7,0,9}, 0);
+              ArrayUtil.remove (new int[] {4,5,6,2,5,8}, 2);
+             ArrayUtil.sum(new int [] {5,8,9,21,1});
+            ArrayUtil.getMax (new int [] {5,69,12,89,25,90,0});
+           ArrayUtil.getMin(new int [] {24,5,15,64,-5,1});
+          ArrayUtil.getAvg(new int [] {5,6,7,8,9,1});
```
Comment (Karlen999) [inline src/arrays/ArrayUtilTest.java:16]: In order not to create a new array for each method, you can create an array in the ArrayUtilTest class and call it.

similar to this

static int[] array = {1, 5, 6, 7, 8};  and call array

File: src/arrays/ArrayUtil.java [hunk 2]
```diff
@@ -1,23 +1,62 @@
...
-
+    int arr[] = new int[size];
+        for (int i = 0; i < size; i++) {
+            Scanner in = new Scanner(System.in);
+            arr[i] = in.nextInt();
+        }
+        for (int i = 0; i < size; i++) {
+            System.out.print(arr[i] + " ");
+        }
+        System.out.println();
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:23]: this **System.out.println();** can be removed

File: src/arrays/ArrayUtil.java [hunk 3]
```diff
@@ -1,23 +1,62 @@
...
+            }
+        }
+        for (int i = 0; i < x; i++) {
+            for (int j = 0; j < y; j++) {
+                System.out.print(arrtwo[i][j] + " ");
+            }
+            System.out.println();
+        }
+        System.out.println();
+
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:42]: done

File: src/arrays/ArrayUtil.java [hunk 4]
```diff
@@ -1,23 +1,62 @@
...
+                l += 2;
+            }
+        }
+        for (int i = 0; i < x; i++) {
+            for (int j = 0; j < y; j++) {
+                System.out.print(arrtwo1[i][j] + " ");
+            }
+            System.out.println();
+        }
+        System.out.println();
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:59]: done

File: src/arrays/ArrayUtil.java [hunk 5]
```diff
@@ -1,23 +1,62 @@
...
     }
 
-    /**
-     * Create two dimensional array with the given sizes, fill it with different values and print it.
-     */
     public static void create2dArray(int x, int y) {
+
+    int arrtwo[][] = new int[x][y];
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:28]: when declaring a variable use variable naming convention (Camel case)

File: src/arrays/ArrayUtil.java [hunk 6]
```diff
@@ -26,8 +65,11 @@ public static void create2dArray(int x, int y) {
      * @return new array.
      */
     public static int[] addFirst(int[] array, int valueToAdd) {
-
-        return null;
+    int arr[] = new int[array.length + 1];
+    arr[0] = valueToAdd;
+        System.arraycopy(array, 0, arr, 1, array.length);
+         System.out.println(Arrays.toString(arr));
+          return arr;
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:72]: done

File: src/arrays/ArrayUtil.java [hunk 7]
```diff
@@ -36,8 +78,19 @@ public static void create2dArray(int x, int y) {
...
-
-        return false;
+     boolean found = false;
+        for (int k = 0; k < array.length; k++) {
+           if (array[k] == value) {
+               found = true;
+                break;
+            }
+        }
+        System.out.println(found);
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:88]: you can return **found** , and the last lines will not be needed ))

File: src/arrays/ArrayUtil.java [hunk 8]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
-
-        return 0;
+    int found = -1;
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == value) {
+                found = i;
+                break;
+            }
+        }
+        System.out.println(found);
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:109]: see the previous comment ))

File: src/arrays/ArrayUtil.java [hunk 9]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
     }
 
     /**
      * Remove the element by the given index.
      * Print new array
      */
     public static void remove(int[] array, int index) {
-
+     int [] newarr = new int[array.length - 1];
+     int newarrl = array.length-(index+1);
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:123]: when declaring a variable use variable naming convention (Camel case)

File: src/arrays/ArrayUtil.java [hunk 10]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
+        System.arraycopy(array, 0, newarr, 0, index);
+         System.arraycopy(array, index + 1, newarr, index,newarrl);
+        System.out.println(Arrays.toString(array));
+           System.out.println(Arrays.toString(newarr));
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:127]: done

File: src/arrays/ArrayUtil.java [hunk 11]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
     /**
      * Calculate and return sum of array's elements.
      */
     public static long sum(int[] array) {
-
+    int sum = 0;
+        for (int j = 0; j < array.length; j++) {
+            sum += array[j];
+        }
+        System.out.println(sum);
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:138]: you can return **sum**

File: src/arrays/ArrayUtil.java [hunk 12]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
         return 0;
     }
 
-
     /**
      * Get the max value from the array.
      */
     public static int getMax(int[] array) {
-
+    int Max = 0;
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:146]: with negative values, it will work incorrectly. (see int max = 0)

File: src/arrays/ArrayUtil.java [hunk 13]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
+        for(int i = 0; i < array.length; i++) {
+            if (array[i] > Max) {
+                Max = array[i];
+            }
+        }
+        System.out.println(Max);
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:152]: return **Max**

File: src/arrays/ArrayUtil.java [hunk 14]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
         return 0;
     }
 
     /**
      * Get the minimum value from the array
      */
     public static void getMin(int[] array) {
-
+     int Min = Arrays.stream(array).min().getAsInt();
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:160]: it is necessary to do tasks using the skills of working with arrays, and not ready-made stream methods.

File: src/arrays/ArrayUtil.java [hunk 15]
```diff
@@ -46,50 +99,78 @@ public static boolean contains(int[] array, int value) {
...
      * Calculate average of array
      */
     public static int getAvg(int[] array) {
-
+     int sum = 0;
+        for (int j = 0; j < array.length; j++) {
+            sum += array[j];
+        }
+     int avg = sum / array.length;
+        System.out.println(avg);
```
Comment (Karlen999) [inline src/arrays/ArrayUtil.java:173]: return **avg**

File: src/arrays/ArrayUtil.java (hunk 14 above)
Comment (goharminasyan) [reply src/arrays/ArrayUtil.java:160]: public static void getMin(int[] array) {
     int Min = array[0];
      for(int i=1; i<array.length; i++){
          if(array[i] < Min){
              Min = array[i];
          }
      }
        System.out.println(Min);
    }

**Comment by Karlen999** [comment]
reviewed

**Pull request closed without merging**